haproxy_exporter_test.go

```diff
@@ -104,8 +104,52 @@ func TestServerWithoutChecks(t *testing.T) {
 		// csvParseFailures
 		t.Errorf("expected %f csv parse failures, got %f", expect, got)
 	}
-	// Suck up the remaining metrics.
-	for _ = range ch {
+
+	got := 0
+	for range ch {
+		got += 1
+	}
+	if expect := len(e.serverMetrics) - 1; got != expect {
+		t.Errorf("expected %d metrics, got %d", expect, got)
+	}
+}
+
+func TestServerBrokenCSV(t *testing.T) {
```

Comment:
Done. I couldn't find a commit immediately, but the haproxy mailing list is the most commonly linked source for HAProxy topics.